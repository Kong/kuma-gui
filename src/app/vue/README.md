---
section: Packages
---
# vue

Automatic `vue` initialization using our projects dependency injection layer.

