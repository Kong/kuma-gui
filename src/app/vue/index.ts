import { createApp } from 'vue'
import {
  createRouter,
  createWebHistory,
} from 'vue-router'

import type { ServiceDefinition } from '@/services/utils'
import { token, createInjections } from '@/services/utils'
import type { Component } from 'vue'
import type { Router, RouteRecordRaw, NavigationGuard } from 'vue-router'

type Token = ReturnType<typeof token>

type VueApp = ReturnType<typeof createApp>
export type PluginDefinition = Parameters<VueApp['use']>
export type ComponentDefinition = [string, Component]

const $ = {
  app: token<(App: Component) => Promise<VueApp>>('vue.app'),
  router: token<Router>('vue.router'),

  components: token('vue.components'),
  plugins: token<PluginDefinition[]>('vue.plugins'),
  routes: token<RouteRecordRaw[]>('vue.routes'),
  routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
  navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
}
export const services = (app: Record<string, Token>): ServiceDefinition[] => {
  return [
    [$.app, {
      service: (
        components: ComponentDefinition[],
        plugins: PluginDefinition[],
      ) => {
        return async (App: Component) => {
          const app = createApp(App)

          plugins.forEach(([...args]) => {
            app.use(...args)
          })

          components.forEach(([name, component]: [string, Component]) => {
            app.component(name, component)
          })

          return app
        }
      },
      arguments: [
        $.components,
        $.plugins,
      ],
    }],
    [$.router, {
      service: (
        env: (str: 'KUMA_BASE_PATH') => string,
        routes: RouteRecordRaw[],
        guards: NavigationGuard[],
      ) => {
        const router = createRouter({
          history: createWebHistory(env('KUMA_BASE_PATH')),
          routes,
        })

        guards.forEach((item) => {
          if (typeof item === 'function') {
            router.beforeEach(item)
          }
        })

        return router
      },
      arguments: [
        app.env,
        $.routes,
        $.navigationGuards,
      ],
    }],

    [token('vue.plugins'), {
      service: (router) => {
        return [
          [router],
        ]
      },
      arguments: [
        $.router,
      ],
      labels: [
        $.plugins,
      ],
    }],

    // TODO(jc): Label decoration. Make it so we can decorate labels
    // Temporarily turn the routesLabel label into routes for decoration
    // purposes
    [$.routes, {
      service: (routes: RouteRecordRaw[]) => routes,
      arguments: [
        $.routesLabel,
      ],
    }],
    //

    // TODO(jc): Make it so we don't need to provide a [] empty default for
    // labels
    [token('application.routes.navigation.guards'), {
      service: () => {
        return []
      },
      labels: [
        $.navigationGuards,
      ],
    }],
  ]
}

export const TOKENS = $
export const [
  useRouter,
] = createInjections(
  TOKENS.router,
)
