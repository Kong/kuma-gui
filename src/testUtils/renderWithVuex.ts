import { render, RenderOptions, RenderResult } from '@testing-library/vue'
import merge from 'lodash/merge'
import { StoreOptions } from 'vuex'
import Vue, { ComponentOptions } from 'vue'
import Store from '@/store'

function renderWithVuex(
  Component: typeof Vue | ComponentOptions<Vue>,
  options: RenderOptions<Vue> = { store: {}, routes: [] },
): RenderResult {
  const store: StoreOptions<any> = Store()

  // Render the component and merge the original store and the custom one
  // provided as a parameter. This way, we can alter some behaviors of the
  // initial implementation.

  return render(Component, { ...options, store: merge(store, options.store), routes: options.routes })
}

export default renderWithVuex
