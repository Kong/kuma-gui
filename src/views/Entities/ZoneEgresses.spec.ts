import { screen } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'
import ZoneEgresses from './ZoneEgresses.vue'
import renderWithVuex from '@/testUtils/renderWithVuex'

jest.mock('@/helpers', () => {
  const originalModule = jest.requireActual('@/helpers')

  return {
    __esModule: true,
    ...originalModule,
    humanReadableDate: (data: string) => data,
    rawReadableDate: (data: string) => data,
  }
})

describe('ZoneEgresses.vue', () => {
  it('renders snapshot when no multizone', async () => {
    const { container } = renderWithVuex(ZoneEgresses, {
      routes: [],
      store: { modules: { config: { state: { clientConfig: { mode: 'global' } } } } },
    })

    await screen.findByText(/ZoneEgressOverview/)

    expect(container).toMatchSnapshot()
  })

  it('renders snapshot when multizone', async () => {
    const { container } = renderWithVuex(ZoneEgresses, {
      routes: [],
      store: { modules: { config: { state: { clientConfig: { mode: 'global' } } } } },
    })

    await screen.findByText(/ZoneEgressOverview/)

    expect(container).toMatchSnapshot()
  })

  it('renders zoneegress insights', async () => {
    renderWithVuex(ZoneEgresses, {
      routes: [],
      store: { modules: { config: { state: { clientConfig: { mode: 'global' } } } } },
    })

    await screen.findByText(/ZoneEgressOverview/)

    await userEvent.click(screen.getByText(/Zone Egress Insights/))
    expect(screen.getByTestId('tab-container')).toMatchSnapshot()
  })
})
