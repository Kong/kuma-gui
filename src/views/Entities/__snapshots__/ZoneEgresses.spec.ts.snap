// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneEgresses.vue renders snapshot when multizone 1`] = `
<div>
  <div
    class="zoneegresses"
  >
    <div
      class="data-overview"
      data-testid="data-overview"
    >
      <div
        class="data-table-controls mb-2"
      >
         
        <button
          class="k-button ml-2 refresh-button small primary"
          data-v-38fa8ecf=""
          type="button"
        >
          <div
            class="refresh-icon"
            data-v-38fa8ecf=""
          >
            <svg
              data-v-38fa8ecf=""
              viewBox="0 0 36 36"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-v-38fa8ecf=""
                fill="#fff"
                fill-rule="nonzero"
              >
                <path
                  d="M18 5.5a12.465 12.465 0 00-8.118 2.995 1.5 1.5 0 001.847 2.363l.115-.095A9.437 9.437 0 0118 8.5l.272.004a9.487 9.487 0 019.07 7.75l.04.246H25a.5.5 0 00-.416.777l4 6a.5.5 0 00.832 0l4-6 .04-.072A.5.5 0 0033 16.5h-2.601l-.017-.15C29.567 10.2 24.294 5.5 18 5.5zM2.584 18.723l-.04.072A.5.5 0 003 19.5h2.6l.018.15C6.433 25.8 11.706 30.5 18 30.5c3.013 0 5.873-1.076 8.118-2.995a1.5 1.5 0 00-1.847-2.363l-.115.095A9.437 9.437 0 0118 27.5l-.272-.004a9.487 9.487 0 01-9.07-7.75l-.041-.246H11a.5.5 0 00.416-.777l-4-6a.5.5 0 00-.832 0l-4 6z"
                  data-v-38fa8ecf=""
                />
              </g>
            </svg>
          </div>
           
          <span
            data-v-38fa8ecf=""
          >
            Refresh
          </span>
        </button>
      </div>
       
      <div
        class="data-overview-content"
      >
        <div
          class="data-overview-table"
        >
          <table
            class="k-table micro-table has-hover is-clickable"
            data-v-bfeabd48=""
          >
            <thead
              data-v-bfeabd48=""
            >
              <tr
                data-v-bfeabd48=""
              >
                <th
                  class=""
                  data-v-bfeabd48=""
                >
                  <!---->
                </th>
                <th
                  class=""
                  data-v-bfeabd48=""
                >
                   Status 
                </th>
                <th
                  class=""
                  data-v-bfeabd48=""
                >
                   Name 
                </th>
              </tr>
            </thead>
            <tbody
              data-v-bfeabd48=""
            >
              <tr
                data-v-bfeabd48=""
              >
                <td
                  data-v-bfeabd48=""
                >
                  <a
                    class="data-table-action-link is-active"
                    data-v-bfeabd48=""
                  >
                    <span
                      class="action-link__active-state"
                      data-v-bfeabd48=""
                    >
                      
              âœ“
              
                      <span
                        class="sr-only"
                        data-v-bfeabd48=""
                      >
                        
                Selected
              
                      </span>
                    </span>
                  </a>
                </td>
                <td
                  data-v-bfeabd48=""
                >
                  <div
                    class="entity-status"
                    data-v-bfeabd48=""
                  >
                    <span
                      class="entity-status__dot"
                      data-v-bfeabd48=""
                    />
                     
                    <span
                      class="entity-status__label"
                      data-v-bfeabd48=""
                    >
                      Online
                    </span>
                  </div>
                </td>
                <td
                  data-v-bfeabd48=""
                >
                  zoneegress-1
                </td>
              </tr>
            </tbody>
          </table>
           
          <div
            class="pagination"
          >
            <!---->
             
            <!---->
          </div>
        </div>
         
        <!---->
         
        <!---->
      </div>
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="tab-container"
      data-testid="tab-container"
    >
      <header
        class="tab__header"
      >
        <div>
          <h3>
             Zone Egress: zoneegress-1
          </h3>
        </div>
         
        <div>
          <div
            data-testid="entity-url-control"
          >
            <div
              data-v-3cb990b6=""
            >
              <button
                class="k-button small secondary"
                data-v-38fa8ecf=""
                data-v-3cb990b6=""
                type="button"
              >
                <svg
                  class="kong-icon kong-icon-externalLink"
                  data-v-38fa8ecf=""
                  data-v-a7c9c1d2=""
                  height="12"
                  role="img"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <title
                    data-v-a7c9c1d2=""
                  >
                    externalLink
                  </title>
                  <g
                    data-v-a7c9c1d2=""
                  >
                    <path
                      d="M8.24308 1.5C7.83269 1.5 7.5 1.16710138 7.5.75c0-.41421356.33823-.75.74308-.75h3.00568C11.6666 0 12 .336341 12 .75123866v3.00567786C12 4.1673102 11.6671 4.5 11.25 4.5c-.41421 0-.75-.3382314-.75-.74308348V2.625L6.53246 6.59254c-.28982.2898217-.76955.2953689-1.06202.0029022l-.06588-.0658844c-.29098-.2909742-.29117-.7679483.0029-1.0620178L9.375 1.5H8.24308zm-6.74753 0C.67089 1.5 0 2.1695784 0 2.99554521v7.50890959C0 11.3291147.66958 12 1.49555 12h7.5089C9.82911 12 10.5 11.3304216 10.5 10.5044548V6.75H9v3.75H1.5V3h3.75V1.5H1.49555z"
                      data-v-a7c9c1d2=""
                      fill="#000"
                      fill-opacity=".25"
                      fill-rule="evenodd"
                    />
                  </g>
                </svg>
                
        Copy URL
      
              </button>
              <transition-stub
                data-v-3cb990b6=""
                name="fade"
              >
                <div
                  class="k-popover"
                  data-v-3cb990b6=""
                  style="width: 200px; display: none;"
                >
                  <!---->
                  <div
                    class="k-popover-content"
                    data-v-3cb990b6=""
                  >
                    <div>
                      <p>
                        URL copied to clipboard!
                      </p>
                    </div>
                  </div>
                </div>
              </transition-stub>
            </div>
          </div>
        </div>
      </header>
       
      <div
        class="tab__content-container"
      >
        <div
          class="k-tabs"
          data-v-a38b664a=""
        >
          <ul
            aria-label="ktabs"
            data-v-a38b664a=""
            role="tablist"
          >
            <li
              aria-controls="panel-0"
              aria-selected="true"
              class="tab-item active"
              data-v-a38b664a=""
              id="overview-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Overview
              </a>
            </li>
            <li
              aria-controls="panel-1"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="insights-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Zone Egress Insights
              </a>
            </li>
            <li
              aria-controls="panel-2"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="xds-configuration-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                XDS Configuration
              </a>
            </li>
            <li
              aria-controls="panel-3"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="envoy-stats-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Stats
              </a>
            </li>
            <li
              aria-controls="panel-4"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="envoy-clusters-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Clusters
              </a>
            </li>
          </ul>
          <div
            aria-labelledby="overview-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-0"
            role="tabpanel"
            tabindex="0"
          >
            <div
              data-v-a38b664a=""
            >
              <div
                class="label-list-content"
              >
                <div
                  class="kong-card noBorder"
                  data-v-0029352b=""
                >
                  <!---->
                  <!---->
                  <div
                    class="k-card-body"
                    data-v-0029352b=""
                  >
                    <div
                      class="label-list__col-wrapper multi-col"
                      data-v-0029352b=""
                    >
                      <div
                        data-v-0029352b=""
                      >
                        <ul
                          data-v-0029352b=""
                        >
                          <li
                            data-v-0029352b=""
                          >
                            <h4
                              data-v-0029352b=""
                            >
                              
                type
              
                            </h4>
                             
                            <p
                              data-v-0029352b=""
                            >
                              
                ZoneEgressOverview
              
                            </p>
                          </li>
                          <li
                            data-v-0029352b=""
                          >
                            <h4
                              data-v-0029352b=""
                            >
                              
                name
              
                            </h4>
                             
                            <p
                              data-v-0029352b=""
                            >
                              
                zoneegress-1
              
                            </p>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
          </div>
          <div
            aria-labelledby="insights-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-1"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
          <div
            aria-labelledby="xds-configuration-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-2"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
          <div
            aria-labelledby="envoy-stats-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-3"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
          <div
            aria-labelledby="envoy-clusters-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-4"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
        </div>
         
        <div>
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ZoneEgresses.vue renders snapshot when no multizone 1`] = `
<div>
  <div
    class="zoneegresses"
  >
    <div
      class="data-overview"
      data-testid="data-overview"
    >
      <div
        class="data-table-controls mb-2"
      >
         
        <button
          class="k-button ml-2 refresh-button small primary"
          data-v-38fa8ecf=""
          type="button"
        >
          <div
            class="refresh-icon"
            data-v-38fa8ecf=""
          >
            <svg
              data-v-38fa8ecf=""
              viewBox="0 0 36 36"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-v-38fa8ecf=""
                fill="#fff"
                fill-rule="nonzero"
              >
                <path
                  d="M18 5.5a12.465 12.465 0 00-8.118 2.995 1.5 1.5 0 001.847 2.363l.115-.095A9.437 9.437 0 0118 8.5l.272.004a9.487 9.487 0 019.07 7.75l.04.246H25a.5.5 0 00-.416.777l4 6a.5.5 0 00.832 0l4-6 .04-.072A.5.5 0 0033 16.5h-2.601l-.017-.15C29.567 10.2 24.294 5.5 18 5.5zM2.584 18.723l-.04.072A.5.5 0 003 19.5h2.6l.018.15C6.433 25.8 11.706 30.5 18 30.5c3.013 0 5.873-1.076 8.118-2.995a1.5 1.5 0 00-1.847-2.363l-.115.095A9.437 9.437 0 0118 27.5l-.272-.004a9.487 9.487 0 01-9.07-7.75l-.041-.246H11a.5.5 0 00.416-.777l-4-6a.5.5 0 00-.832 0l-4 6z"
                  data-v-38fa8ecf=""
                />
              </g>
            </svg>
          </div>
           
          <span
            data-v-38fa8ecf=""
          >
            Refresh
          </span>
        </button>
      </div>
       
      <div
        class="data-overview-content"
      >
        <div
          class="data-overview-table"
        >
          <table
            class="k-table micro-table has-hover is-clickable"
            data-v-bfeabd48=""
          >
            <thead
              data-v-bfeabd48=""
            >
              <tr
                data-v-bfeabd48=""
              >
                <th
                  class=""
                  data-v-bfeabd48=""
                >
                  <!---->
                </th>
                <th
                  class=""
                  data-v-bfeabd48=""
                >
                   Status 
                </th>
                <th
                  class=""
                  data-v-bfeabd48=""
                >
                   Name 
                </th>
              </tr>
            </thead>
            <tbody
              data-v-bfeabd48=""
            >
              <tr
                data-v-bfeabd48=""
              >
                <td
                  data-v-bfeabd48=""
                >
                  <a
                    class="data-table-action-link is-active"
                    data-v-bfeabd48=""
                  >
                    <span
                      class="action-link__active-state"
                      data-v-bfeabd48=""
                    >
                      
              âœ“
              
                      <span
                        class="sr-only"
                        data-v-bfeabd48=""
                      >
                        
                Selected
              
                      </span>
                    </span>
                  </a>
                </td>
                <td
                  data-v-bfeabd48=""
                >
                  <div
                    class="entity-status"
                    data-v-bfeabd48=""
                  >
                    <span
                      class="entity-status__dot"
                      data-v-bfeabd48=""
                    />
                     
                    <span
                      class="entity-status__label"
                      data-v-bfeabd48=""
                    >
                      Online
                    </span>
                  </div>
                </td>
                <td
                  data-v-bfeabd48=""
                >
                  zoneegress-1
                </td>
              </tr>
            </tbody>
          </table>
           
          <div
            class="pagination"
          >
            <!---->
             
            <!---->
          </div>
        </div>
         
        <!---->
         
        <!---->
      </div>
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="tab-container"
      data-testid="tab-container"
    >
      <header
        class="tab__header"
      >
        <div>
          <h3>
             Zone Egress: zoneegress-1
          </h3>
        </div>
         
        <div>
          <div
            data-testid="entity-url-control"
          >
            <div
              data-v-3cb990b6=""
            >
              <button
                class="k-button small secondary"
                data-v-38fa8ecf=""
                data-v-3cb990b6=""
                type="button"
              >
                <svg
                  class="kong-icon kong-icon-externalLink"
                  data-v-38fa8ecf=""
                  data-v-a7c9c1d2=""
                  height="12"
                  role="img"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <title
                    data-v-a7c9c1d2=""
                  >
                    externalLink
                  </title>
                  <g
                    data-v-a7c9c1d2=""
                  >
                    <path
                      d="M8.24308 1.5C7.83269 1.5 7.5 1.16710138 7.5.75c0-.41421356.33823-.75.74308-.75h3.00568C11.6666 0 12 .336341 12 .75123866v3.00567786C12 4.1673102 11.6671 4.5 11.25 4.5c-.41421 0-.75-.3382314-.75-.74308348V2.625L6.53246 6.59254c-.28982.2898217-.76955.2953689-1.06202.0029022l-.06588-.0658844c-.29098-.2909742-.29117-.7679483.0029-1.0620178L9.375 1.5H8.24308zm-6.74753 0C.67089 1.5 0 2.1695784 0 2.99554521v7.50890959C0 11.3291147.66958 12 1.49555 12h7.5089C9.82911 12 10.5 11.3304216 10.5 10.5044548V6.75H9v3.75H1.5V3h3.75V1.5H1.49555z"
                      data-v-a7c9c1d2=""
                      fill="#000"
                      fill-opacity=".25"
                      fill-rule="evenodd"
                    />
                  </g>
                </svg>
                
        Copy URL
      
              </button>
              <transition-stub
                data-v-3cb990b6=""
                name="fade"
              >
                <div
                  class="k-popover"
                  data-v-3cb990b6=""
                  style="width: 200px; display: none;"
                >
                  <!---->
                  <div
                    class="k-popover-content"
                    data-v-3cb990b6=""
                  >
                    <div>
                      <p>
                        URL copied to clipboard!
                      </p>
                    </div>
                  </div>
                </div>
              </transition-stub>
            </div>
          </div>
        </div>
      </header>
       
      <div
        class="tab__content-container"
      >
        <div
          class="k-tabs"
          data-v-a38b664a=""
        >
          <ul
            aria-label="ktabs"
            data-v-a38b664a=""
            role="tablist"
          >
            <li
              aria-controls="panel-0"
              aria-selected="true"
              class="tab-item active"
              data-v-a38b664a=""
              id="overview-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Overview
              </a>
            </li>
            <li
              aria-controls="panel-1"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="insights-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Zone Egress Insights
              </a>
            </li>
            <li
              aria-controls="panel-2"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="xds-configuration-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                XDS Configuration
              </a>
            </li>
            <li
              aria-controls="panel-3"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="envoy-stats-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Stats
              </a>
            </li>
            <li
              aria-controls="panel-4"
              aria-selected="false"
              class="tab-item"
              data-v-a38b664a=""
              id="envoy-clusters-tab"
              role="tab"
              tabindex="0"
            >
              <a
                class="tab-link"
                data-v-a38b664a=""
              >
                Clusters
              </a>
            </li>
          </ul>
          <div
            aria-labelledby="overview-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-0"
            role="tabpanel"
            tabindex="0"
          >
            <div
              data-v-a38b664a=""
            >
              <div
                class="label-list-content"
              >
                <div
                  class="kong-card noBorder"
                  data-v-0029352b=""
                >
                  <!---->
                  <!---->
                  <div
                    class="k-card-body"
                    data-v-0029352b=""
                  >
                    <div
                      class="label-list__col-wrapper multi-col"
                      data-v-0029352b=""
                    >
                      <div
                        data-v-0029352b=""
                      >
                        <ul
                          data-v-0029352b=""
                        >
                          <li
                            data-v-0029352b=""
                          >
                            <h4
                              data-v-0029352b=""
                            >
                              
                type
              
                            </h4>
                             
                            <p
                              data-v-0029352b=""
                            >
                              
                ZoneEgressOverview
              
                            </p>
                          </li>
                          <li
                            data-v-0029352b=""
                          >
                            <h4
                              data-v-0029352b=""
                            >
                              
                name
              
                            </h4>
                             
                            <p
                              data-v-0029352b=""
                            >
                              
                zoneegress-1
              
                            </p>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
          </div>
          <div
            aria-labelledby="insights-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-1"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
          <div
            aria-labelledby="xds-configuration-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-2"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
          <div
            aria-labelledby="envoy-stats-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-3"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
          <div
            aria-labelledby="envoy-clusters-tab"
            class="tab-container"
            data-v-a38b664a=""
            id="panel-4"
            role="tabpanel"
            tabindex="0"
          >
            <!---->
          </div>
        </div>
         
        <div>
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ZoneEgresses.vue renders zoneegress insights 1`] = `
<div
  class="tab-container"
  data-testid="tab-container"
>
  <header
    class="tab__header"
  >
    <div>
      <h3>
         Zone Egress: zoneegress-1
      </h3>
    </div>
     
    <div>
      <div
        data-testid="entity-url-control"
      >
        <div
          data-v-3cb990b6=""
        >
          <button
            class="k-button small secondary"
            data-v-38fa8ecf=""
            data-v-3cb990b6=""
            type="button"
          >
            <svg
              class="kong-icon kong-icon-externalLink"
              data-v-38fa8ecf=""
              data-v-a7c9c1d2=""
              height="12"
              role="img"
              viewBox="0 0 16 16"
              width="12"
            >
              <title
                data-v-a7c9c1d2=""
              >
                externalLink
              </title>
              <g
                data-v-a7c9c1d2=""
              >
                <path
                  d="M8.24308 1.5C7.83269 1.5 7.5 1.16710138 7.5.75c0-.41421356.33823-.75.74308-.75h3.00568C11.6666 0 12 .336341 12 .75123866v3.00567786C12 4.1673102 11.6671 4.5 11.25 4.5c-.41421 0-.75-.3382314-.75-.74308348V2.625L6.53246 6.59254c-.28982.2898217-.76955.2953689-1.06202.0029022l-.06588-.0658844c-.29098-.2909742-.29117-.7679483.0029-1.0620178L9.375 1.5H8.24308zm-6.74753 0C.67089 1.5 0 2.1695784 0 2.99554521v7.50890959C0 11.3291147.66958 12 1.49555 12h7.5089C9.82911 12 10.5 11.3304216 10.5 10.5044548V6.75H9v3.75H1.5V3h3.75V1.5H1.49555z"
                  data-v-a7c9c1d2=""
                  fill="#000"
                  fill-opacity=".25"
                  fill-rule="evenodd"
                />
              </g>
            </svg>
            
        Copy URL
      
          </button>
          <transition-stub
            data-v-3cb990b6=""
            name="fade"
          >
            <div
              class="k-popover"
              data-v-3cb990b6=""
              style="width: 200px; display: none;"
            >
              <!---->
              <div
                class="k-popover-content"
                data-v-3cb990b6=""
              >
                <div>
                  <p>
                    URL copied to clipboard!
                  </p>
                </div>
              </div>
            </div>
          </transition-stub>
        </div>
      </div>
    </div>
  </header>
   
  <div
    class="tab__content-container"
  >
    <div
      class="k-tabs"
      data-v-a38b664a=""
    >
      <ul
        aria-label="ktabs"
        data-v-a38b664a=""
        role="tablist"
      >
        <li
          aria-controls="panel-0"
          aria-selected="false"
          class="tab-item"
          data-v-a38b664a=""
          id="overview-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            Overview
          </a>
        </li>
        <li
          aria-controls="panel-1"
          aria-selected="true"
          class="tab-item active"
          data-v-a38b664a=""
          id="insights-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            Zone Egress Insights
          </a>
        </li>
        <li
          aria-controls="panel-2"
          aria-selected="false"
          class="tab-item"
          data-v-a38b664a=""
          id="xds-configuration-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            XDS Configuration
          </a>
        </li>
        <li
          aria-controls="panel-3"
          aria-selected="false"
          class="tab-item"
          data-v-a38b664a=""
          id="envoy-stats-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            Stats
          </a>
        </li>
        <li
          aria-controls="panel-4"
          aria-selected="false"
          class="tab-item"
          data-v-a38b664a=""
          id="envoy-clusters-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-a38b664a=""
          >
            Clusters
          </a>
        </li>
      </ul>
      <div
        aria-labelledby="overview-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-0"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="insights-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-1"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="kong-card noBorder"
          data-v-0029352b=""
          data-v-a38b664a=""
        >
          <!---->
          <!---->
          <div
            class="k-card-body"
            data-v-0029352b=""
          >
            <ul
              class="accordion"
              data-v-0029352b=""
            >
              <li
                class="relative border-b py-2 active"
              >
                <button
                  aria-expanded="true"
                  class="accordion-item-header"
                >
                  <h4
                    class="text-lg font-medium"
                  >
                    <span
                      class="color-green-400"
                    >
                      
    Connect time: 2021-07-13T08:41:04.556796688Z
  
                    </span>
                     
                    <!---->
                  </h4>
                </button>
                 
                <transition-stub
                  name="accordion"
                >
                  <div
                    class="px-4"
                  >
                    <div>
                      <div>
                        <h5
                          class="overview-tertiary-title"
                        >
                          
      General Information:
    
                        </h5>
                         
                        <ul>
                          <!---->
                           
                          <li>
                            <strong>
                              Control Plane Instance ID:
                            </strong>
                            Â 
        
                            <span
                              class="mono"
                            >
                              kuma-control-plane-84f5589874-nmspq-0867
                            </span>
                          </li>
                           
                          <li>
                            <strong>
                              Last Connected:
                            </strong>
                            Â 
        2021-07-13T08:41:04.556796688Z
      
                          </li>
                           
                          <!---->
                        </ul>
                      </div>
                       
                      <div>
                        <ul
                          class="overview-stat-grid"
                        >
                          <li>
                            <h6
                              class="overview-tertiary-title"
                            >
                              
          Cds:
        
                            </h6>
                             
                            <ul>
                              <li>
                                <strong>
                                  Responses Sent:
                                </strong>
                                Â 
            
                                <span
                                  class="mono"
                                >
                                  3
                                </span>
                              </li>
                              <li>
                                <strong>
                                  Responses Acknowledged:
                                </strong>
                                Â 
            
                                <span
                                  class="mono"
                                >
                                  3
                                </span>
                              </li>
                            </ul>
                          </li>
                          <li>
                            <h6
                              class="overview-tertiary-title"
                            >
                              
          Eds:
        
                            </h6>
                             
                            <ul>
                              <li>
                                <strong>
                                  Responses Sent:
                                </strong>
                                Â 
            
                                <span
                                  class="mono"
                                >
                                  2
                                </span>
                              </li>
                              <li>
                                <strong>
                                  Responses Acknowledged:
                                </strong>
                                Â 
            
                                <span
                                  class="mono"
                                >
                                  3
                                </span>
                              </li>
                            </ul>
                          </li>
                          <li>
                            <h6
                              class="overview-tertiary-title"
                            >
                              
          Lds:
        
                            </h6>
                             
                            <ul>
                              <li>
                                <strong>
                                  Responses Sent:
                                </strong>
                                Â 
            
                                <span
                                  class="mono"
                                >
                                  3
                                </span>
                              </li>
                              <li>
                                <strong>
                                  Responses Acknowledged:
                                </strong>
                                Â 
            
                                <span
                                  class="mono"
                                >
                                  3
                                </span>
                              </li>
                            </ul>
                          </li>
                          <li>
                            <h6
                              class="overview-tertiary-title"
                            >
                              
          Rds:
        
                            </h6>
                             
                            <ul />
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </transition-stub>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="xds-configuration-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-2"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="envoy-stats-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-3"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="envoy-clusters-tab"
        class="tab-container"
        data-v-a38b664a=""
        id="panel-4"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
    </div>
     
    <div>
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
